### DISCUSSION TITLE
Running Terraria from .AppImage

### DISCUSSION THREAD

**Discussion by contrarybaton60** [ivan-hc/AM#1004] [General]
though not related to AM, I just wanna say that I manage to run Terraria as a .AppImage

https://codeberg.org/PeacefulWilliam/TerrariaOnAppImage/

**Comment by ivan-hc** [comment]
sorry, it seems that the URL is broken

> https://codeberg.org/PeacefulWilliam/TerrariaOnAppImage/

**Comment by contrarybaton60** [comment]
oops.. I still have it as private

**Comment by contrarybaton60** [comment]
should work now

**Comment by ivan-hc** [comment]
where is the appimage?

**Comment by contrarybaton60** [comment]
the readme is a tutorial on how to make an appimage of terraria, I didn't include one for obvious reasons

**Comment by contrarybaton60** [comment]
here's footage of me running it: https://www.youtube.com/watch?v=JaYRzENpcwQ

**Comment by ivan-hc** [comment]
well, I also I created an AppImage of Diablo 2 LOD, times ago, including wine... but being commercial software, I prevent to share this stuff in AM... anyway, well done!

**Comment by contrarybaton60** [comment]
I can probably create one of [Chocolate Doom](https://www.chocolate-doom.org/wiki/index.php/Chocolate_Doom) if I had more time (presumably from .deb packages)

**Comment by Samueru-sama** [comment]
You can turn this into a script where you only have to provide the binary and from there it automatically does the rest. Also is terraria a single binary? I notice you went straight to appimagetool. 

I also started making appimages by first using portable binaries hehe.

**Reply by contrarybaton60** [reply]
> You can turn this into a script where you only have to provide the binary and from there it automatically does the rest. Also is terraria a single binary? I notice you went straight to appimagetool.
> 
> I also started making appimages by first using portable binaries hehe.

![image](https://github.com/user-attachments/assets/69a4f127-7df1-45f6-b72f-35c0b39a7211)
It contains a binary, and a bunch of windows dlls (and it doesn't rely on wine, I don't have it installed)

as for the script, I can probably make it, I just need to learn a little more making them.

**Comment by Azathothas** [comment]
Interesting..
I haven't found many Appimage version of games as there's not many native Linux games..
I will try building it locally.
Don't know what the stance of Ivan or your (@contrarybaton60) stance on piracy is...
Perhaps it's best to avoid such discussions here..


Also checkout https://github.com/ruanformigoni/gameimage , it's much fitting format Incase the portable binary needs to have wine/proton.

**Reply by contrarybaton60** [reply]
SuperTux and Prism Launcher uses AppImage for their builds.

Never heard of GameImage, I'll likely won't use it as many (linux native, and those from GOG) games don't require dependencies (and those that are likely are included in your distro anyways... I hope)
